extensions/amp-analytics/0.1/events.js

```diff
@@ -389,6 +395,43 @@ export class VisibilityTracker extends EventTracker {
   }
 
   /**
+   * @param {string=} waitForSpec
+   * @param {string=} selector
+   * @param {Element=} element
+   * @return {?Promise}
+   * @visibleForTesting
+   */
+  getReadyPromise(waitForSpec, selector, element) {
+    if (waitForSpec == WAITFOR_EVENTS_LIST.NONE || !selector) {
+      // Always return null if specific asked waitFor NONE, or no selector
+      return null;
+    }
+    if (!element) {
+      // If selector is :root or :host, always wait for root ini-load
+      return this.iniLoadTracker_.getRootSignal();
+    }
+    if (!waitForSpec) {
+      // Default behavior for selector select AMP element is to wait for
+      // AMP element's ini-load
+      waitForSpec = WAITFOR_EVENTS_LIST.INI_LOAD;
+    }
+    if (typeof element.signals != 'function') {
+      return null;
+    }
+    switch (waitForSpec) {
+      case WAITFOR_EVENTS_LIST.INI_LOAD:
+        return this.iniLoadTracker_.getElementSignal(element);
+        break;
+      case WAITFOR_EVENTS_LIST.RENDER_START:
+        // may never resolve.
+        return element.signals().whenSignal(WAITFOR_EVENTS_LIST.RENDER_START);
```

Comment:
for `ini-load` I copied code from `iniLoadTracker`. I think it makes `waitFor: ini-load` a little bit confusing.
With RENDER_START we strictly wait for `render-start`. However with INI_LOAD, we can proceed even if `ini-load` has never been sent. 
It's easy for people to assume `waitFor: ini-load` will only track visibility after `ini-load` signal. However in the case where 'ini-load' is never received, visibility tracking can still happen on `#layoutCallback` fail.
@dvoytenko What do you think?